### PR TITLE
Update rust-miniscript to 8fcbeb12 (right before bitcoin 0.29 update)

### PR DESCRIPTION
**Pull request by apoelstra**
This is already a nontrivial amount of work which needs to be done prior to updating rust-bitcoin and rust-elements.

I think, before doing the next step, we should port the new `Locktime` and `Sequence` data structures from rust-bitcoin to rust-elements, and do a rust-elements 0.21 release. That way we can keep all the elements-miniscript code consistent with the rust-miniscript code (which very heavily makes use of locktimes).

### CODE REVIEW

File: src/descriptor/mod.rs [hunk 1]
```diff
@@ -775,57 +772,83 @@ impl<Pk: MiniscriptKey, T: Extension> ForEachKey<Pk> for Descriptor<Pk, T> {
...
+    pub fn derive(&self, index: u32) -> Descriptor<DefiniteDescriptorKey, Ext> {
+        self.at_derivation_index(index)
+    }
+
+    /// Convert all the public keys in the descriptor to [`bitcoin::PublicKey`] by deriving them or
+    /// otherwise converting them. All [`bitcoin::XOnlyPublicKey`]s are converted to by adding a
+    /// default(0x02) y-coordinate.
     ///
-    /// # Examples
+    /// This is a shorthand for:
```
Comment (sanket1729) [inline src/descriptor/mod.rs:833]: This statement is incomplete. Perhaps, an upstream bug

**Review by sanket1729 — APPROVED** [review]
ACK 5093ba451d435f93f0d638415672a93c4640312e. Seemed simpler to review the entire diff at once. Carefully look ed at changes touching

**Comment by sanket1729** [comment]
Merging this first. Other PRs are easy to rebase.

File: src/descriptor/mod.rs (hunk 1 above)
Comment (apoelstra) [reply src/descriptor/mod.rs:833]: Yeah, looks like it's an upstream thing. We should fix it there.

**Pull request merged**